src/main/java/org/folio/validate/definition/SelectableField.java

```diff
@@ -0,0 +1,92 @@
+package org.folio.validate.definition;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.Validate;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import org.folio.rest.jaxrs.model.CustomField;
+import org.folio.rest.jaxrs.model.SelectField;
+
+@Component
+public abstract class SelectableField {
+
+  @Value("${custom.fields.definition.name.length}")
+  private int NAME_LENGTH_LIMIT;
+
+  @Value("${custom.fields.definition.single.default.size}")
+  private int SINGLE_DEFAULT_SIZE;
+
+  @Value("#{${custom.fields.definition.option.size}}")
+  private Map<String, Integer> optionsSizeMap;
+
+  void validateOptions(CustomField fieldDefinition) {
+    final SelectField selectField = fieldDefinition.getSelectField();
+
+    if (Objects.nonNull(selectField) && Objects.nonNull(selectField.getOptions())) {
+
+      final List<String> values = selectField.getOptions().getValues();
+      final Integer maxOptionSize = optionsSizeMap.get(fieldDefinition.getType().value());
+
+      if(!values.isEmpty()) {
+        Validate.isTrue(values.size() <= maxOptionSize,
+          "The max option size for '" + fieldDefinition.getType() + "' custom field type is %s", maxOptionSize);
+
+        Validate.isTrue(values.stream().allMatch(value -> StringUtils.isNotBlank(value) && value.length() <= NAME_LENGTH_LIMIT),
+          "The option name cannot be blank or have more than %s characters", NAME_LENGTH_LIMIT);
+      }
+    }
+  }
+
+  void validateDefaults(CustomField fieldDefinition) {
+    final SelectField selectField = fieldDefinition.getSelectField();
+
+    if (Objects.nonNull(selectField) && Objects.nonNull(selectField.getDefaults())) {
+
+      List<String> defaults = selectField.getDefaults();
+
+      if (!defaults.isEmpty()) {
```

Comment:
Validation should work even if defaults is empty, so I think this condition can be removed

Reply:
removed